staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -353,8 +354,14 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 	a.cachedCreds = newCreds
 	// Only close all connections when TLS cert rotates. Token rotation doesn't
 	// need the extra noise.
-	if a.onRotate != nil && oldCreds != nil && !reflect.DeepEqual(oldCreds.cert, a.cachedCreds.cert) {
-		a.onRotate()
+	if oldCreds != nil && !reflect.DeepEqual(oldCreds.cert, a.cachedCreds.cert) {
+		if oldCreds.cert != nil { // Can be nil if the exec auth plugin only returned token auth.
+			metrics.ClientCertRotationAge.Observe(time.Now().Sub(oldCreds.cert.Leaf.NotBefore))
```

Comment:
Why only report this when rotation happens?
Can we report cert age every time in `getCreds` instead?

Reply:
This metric is observing the age of the last certificate right before it is thrown out. It seems appropriate that it is a simple histogram that _ideally_ reports a data point every week or so.

Reply:
Ah, I thought this was reporting the age of _new_ cert, not the _old_ one.
Makes sense now.